docs/tutorials/etl.md

```diff
@@ -0,0 +1,518 @@
+are not keyed, or both are keyed, and if they are both keyed, the key values have to be the same. In
+this case the streams are both of type `DataStream<String>`, and both streams are keyed by the
+string. As you will see below, this `RichCoFlatMapFunction` is storing a Boolean value in keyed
+state, and this Boolean is shared by the two streams.
+
+{% highlight java %}
+public static class ControlFunction extends RichCoFlatMapFunction<String, String, String> {
+    private ValueState<Boolean> blocked;
+      
+    @Override
+    public void open(Configuration config) {
+        blocked = getRuntimeContext().getState(new ValueStateDescriptor<>("blocked", Boolean.class));
+    }
+      
+    @Override
+    public void flatMap1(String control_value, Collector<String> out) throws Exception {
+        blocked.update(Boolean.TRUE);
+    }
+      
+    @Override
+    public void flatMap2(String data_value, Collector<String> out) throws Exception {
+        if (blocked.value() == null) {
+            out.collect(data_value);
+        }
+    }
+}
+{% endhighlight %}
+
+A `RichCoFlatMapFunction` is a kind of `FlatMapFunction` that can be applied to a pair of connected
+streams, and it has access to the rich function interface. This means that it can be made stateful.
+
+The `blocked` Boolean is being used to remember the keys (words, in this case) that have been
+mentioned on the `control` stream, and those words are being filtered out of the `streamOfWords`
+stream. This is _keyed_ state, and it is shared between the two streams, which is why the two
+streams have to share the same keyspace.
+
+`flatMap1` and `flatMap2` are called by the Flink runtime with elements from each of the two
+connected streams -- in our case, elements from the `control` stream are passed into `flatMap1`, and
+elements from `streamOfWords` are passed into `flatMap2`. This was determined by the order in which
+the two streams are connected with `control.connect(datastreamOfWords)`. 
+
+It is important to recognize that you have no control over the order in which the `flatMap1` and
+`flatMap2` callbacks are called. These two input streams are racing against each other, and the
+Flink runtime will do what it wants to regarding consuming events from one stream or the other. In
+cases where timing and/or ordering matter, you may find it necessary to buffer events in managed
+Flink state until your application is ready to process them. (Note: if you are truly desperate, it
+is possible to exert some limited control over the order in which a two-input operator consumes its
+inputs by using a custom Operator that implements the
+[InputSelectable](https://ci.apache.org/projects/flink/flink-docs-stable/api/java/org/apache/flink/streaming/api/operators/InputSelectable.html)
+interface.)
+
+{% top %}
+
+## Hands-on
+
+The hands-on exercise that goes with this section is the [Rides and Fares
+Exercise](https://github.com/apache/flink-training/tree/master/rides-and-fares).
```

Comment:
the operations playground instructions at [1] try to be clever and use a URL that matches the documentation version but messes up with `master`...it would, however, be nice to have something like this (if not now, then please open a ticket for it)

[1] https://ci.apache.org/projects/flink/flink-docs-master/getting-started/docker-playgrounds/flink-operations-playground.html#starting-the-playground

Reply:
Ok, I figured out how to do that.